app/services/forest_liana/filters_parser.rb

```diff
@@ -47,28 +47,47 @@ def parse_aggregation(node)
     end
 
     def parse_condition(condition)
+      where = parse_condition_without_smart_field(condition)
+
+      if SchemaUtils.is_smart_field(@resource, field)
+        schema = ForestLiana.schema_for_resource(@resource)
+        field_schema = schema.fields.find do |field|
+          field.try(:name) == field_name
+        end
+
+        unless field_schema.try(:[], :filter)
+          raise ForestLiana::Errors::ExpectedError.new(500, "Server Error", "method filter on smart field '#{field_name}' not found")
```

Comment:
[MISC] You're raising an expected error but it's a 500, a bit counter intuitive don't you think?

Reply:
The name of the error was bad. I renamed it to HTTPError